<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.554">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Jan Emmanuel Samson">

<title>Ab initio Protein Structure Prediction with AlphaFold3</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="main_files/libs/clipboard/clipboard.min.js"></script>
<script src="main_files/libs/quarto-html/quarto.js"></script>
<script src="main_files/libs/quarto-html/popper.min.js"></script>
<script src="main_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="main_files/libs/quarto-html/anchor.min.js"></script>
<link href="main_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="main_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="main_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="main_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="main_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="main_files/libs/quarto-diagram/mermaid.min.js"></script>
<script src="main_files/libs/quarto-diagram/mermaid-init.js"></script>
<link href="main_files/libs/quarto-diagram/mermaid.css" rel="stylesheet">

  <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
  <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-chtml-full.js" type="text/javascript"></script>

<script type="text/javascript">
const typesetMath = (el) => {
  if (window.MathJax) {
    // MathJax Typeset
    window.MathJax.typeset([el]);
  } else if (window.katex) {
    // KaTeX Render
    var mathElements = el.getElementsByClassName("math");
    var macros = [];
    for (var i = 0; i < mathElements.length; i++) {
      var texText = mathElements[i].firstChild;
      if (mathElements[i].tagName == "SPAN") {
        window.katex.render(texText.data, mathElements[i], {
          displayMode: mathElements[i].classList.contains('display'),
          throwOnError: false,
          macros: macros,
          fleqn: false
        });
      }
    }
  }
}
window.Quarto = {
  typesetMath
};
</script>

</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#alphafold3" id="toc-alphafold3" class="nav-link active" data-scroll-target="#alphafold3">AlphaFold3</a>
  <ul class="collapse">
  <li><a href="#claims" id="toc-claims" class="nav-link" data-scroll-target="#claims">Claims</a></li>
  </ul></li>
  <li><a href="#alphafold-server" id="toc-alphafold-server" class="nav-link" data-scroll-target="#alphafold-server">AlphaFold Server</a>
  <ul class="collapse">
  <li><a href="#navigate-to-the-server" id="toc-navigate-to-the-server" class="nav-link" data-scroll-target="#navigate-to-the-server">Navigate to the Server</a></li>
  <li><a href="#login-with-gmail" id="toc-login-with-gmail" class="nav-link" data-scroll-target="#login-with-gmail">Login with Gmail</a></li>
  <li><a href="#explore-the-server-page" id="toc-explore-the-server-page" class="nav-link" data-scroll-target="#explore-the-server-page">Explore the Server Page</a></li>
  <li><a href="#provide-input-sequences" id="toc-provide-input-sequences" class="nav-link" data-scroll-target="#provide-input-sequences">Provide Input Sequences</a></li>
  <li><a href="#submit-job-and-view-results" id="toc-submit-job-and-view-results" class="nav-link" data-scroll-target="#submit-job-and-view-results">Submit Job and View Results</a></li>
  </ul></li>
  <li><a href="#json-format-for-running-jobs" id="toc-json-format-for-running-jobs" class="nav-link" data-scroll-target="#json-format-for-running-jobs">JSON Format for Running Jobs</a>
  <ul class="collapse">
  <li><a href="#job-name-seeds-and-sequences" id="toc-job-name-seeds-and-sequences" class="nav-link" data-scroll-target="#job-name-seeds-and-sequences">Job name, seeds and sequences</a></li>
  <li><a href="#entity-types" id="toc-entity-types" class="nav-link" data-scroll-target="#entity-types">Entity types</a></li>
  <li><a href="#json-structure" id="toc-json-structure" class="nav-link" data-scroll-target="#json-structure">JSON Structure</a></li>
  </ul></li>
  <li><a href="#frequently-asked-questions" id="toc-frequently-asked-questions" class="nav-link" data-scroll-target="#frequently-asked-questions">Frequently Asked Questions</a></li>
  </ul>
</nav>
</div>
<main class="content page-columns page-full" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><em>Ab initio</em> Protein Structure Prediction with AlphaFold3</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Jan Emmanuel Samson </p>
          </div>
  </div>
    
  
    
  </div>
  


</header>


<section id="alphafold3" class="level1 page-columns page-full">
<h1>AlphaFold3</h1>
<ul>
<li>a machine learning model used to predict the joint structure of complexes
<ul>
<li>proteins</li>
<li>nucleic acids</li>
<li>small molecules</li>
<li>ions</li>
<li>modified residues</li>
</ul></li>
</ul>

<div class="no-row-height column-margin column-container"><div class="">
<p>The paper describing AlphaFold3 has not been published, but a draft is available <a href="https://www.nature.com/articles/s41586-024-07487-w">here</a>.</p>
</div></div><table class="table-hover table-striped table">
<caption>Feature comparison of AlphaFold2 and AlphaFold3 models.</caption>
<thead>
<tr class="header">
<th style="text-align: left;">Criterion</th>
<th style="text-align: center;">AlphaFold2</th>
<th style="text-align: center;">AlphaFold3</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;">MSA processing</td>
<td style="text-align: center;">Evoformer</td>
<td style="text-align: center;">Pairformer</td>
</tr>
<tr class="even">
<td style="text-align: left;">Structure prediction</td>
<td style="text-align: center;">Structure module</td>
<td style="text-align: center;">Diffusion module</td>
</tr>
<tr class="odd">
<td style="text-align: left;">Limitations</td>
<td style="text-align: center;">Proteins</td>
<td style="text-align: center;">Complexes</td>
</tr>
<tr class="even">
<td style="text-align: left;">Server provision</td>
<td style="text-align: center;">No</td>
<td style="text-align: center;">Yes</td>
</tr>
</tbody>
</table>
<section id="claims" class="level2">
<h2 class="anchored" data-anchor-id="claims">Claims</h2>
<ul>
<li>greater accuracy on predicting protein-ligand interactions as compared to <em>state of the art</em> docking tools</li>
<li>higher accuracy on protein-nucleic acid interactions than specific predictors</li>
<li>significantly higher antibody-antigen prediction accuracy than AlphaFold-Multimer v2.3</li>
</ul>
</section>
</section>
<section id="alphafold-server" class="level1 page-columns page-full">
<h1>AlphaFold Server</h1>
<p>According to their official page:</p>
<blockquote class="blockquote">
<p>AlphaFold Server is a web-service that can generate highly accurate biomolecular structure predictions containing proteins, DNA, RNA, ligands, ions, and also model chemical modifications for proteins and nucleic acids in one platform. It’s powered by the newest AlphaFold3 model.</p>
</blockquote>
<section id="navigate-to-the-server" class="level2">
<h2 class="anchored" data-anchor-id="navigate-to-the-server">Navigate to the Server</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Step 1
</div>
</div>
<div class="callout-body-container callout-body">
<p>Navigate to the AlphaFold Server website by clicking <a href="https://golgi.sandbox.google.com/">here</a>.</p>
</div>
</div>
<p>The website provides a short description of the server’s functions, as well as a demo:</p>
<div class="quarto-video ratio ratio-16x9"><iframe data-external="1" src="https://www.youtube.com/embed/9ufplEgtq8w" title="" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe></div>
<p>In order to gain access to the server, we must log in with our email account.</p>
<p><img src="/assets/login.png" class="img-fluid"></p>
</section>
<section id="login-with-gmail" class="level2">
<h2 class="anchored" data-anchor-id="login-with-gmail">Login with Gmail</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Step 2
</div>
</div>
<div class="callout-body-container callout-body">
<p>Log-in with you email by clicking on <code>Continue with Google</code>.</p>
</div>
</div>
<p>You’ll be redirected to the server. In the navigation bar, we can see the following components:</p>
<ol type="1">
<li><strong>Server</strong> - main page for running simulations</li>
<li><strong>About</strong> - a few details on how the server works</li>
<li><strong>FAQs</strong> - frequently asked questions; consult here prior to reporting an issue</li>
<li><strong>Feedback</strong> - report an issue or send some suggestions</li>
<li><strong>Toggle</strong> nightmode - toggle color theme</li>
<li><strong>Account</strong> - access settings; logout</li>
</ol>
</section>
<section id="explore-the-server-page" class="level2">
<h2 class="anchored" data-anchor-id="explore-the-server-page">Explore the Server Page</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Step 3
</div>
</div>
<div class="callout-body-container callout-body">
<p>Make sure you’re currently at the <strong>Server</strong> page. Explore its contents.</p>
</div>
</div>
<p><img src="/assets/server.png" class="img-fluid"></p>
<p>It is important to note that you can send a limited number of requests to the server, as indicated by the <em>Remaining jobs</em> count. For each job, the input structure is constrained by a limit of 5,000 tokens. Tokens are computed as follows:</p>
<ul>
<li><strong>Protein</strong>: 1 token per standard amino acid residue</li>
<li><strong>DNA</strong>, <strong>RNA</strong>: 1 token per input nucleotide base</li>
<li><strong>Ligands</strong>: 1 token per atom in the ligand</li>
<li><strong>Ions</strong>: 1 token per ion</li>
<li><strong>Modification</strong>s (excluding glycans): 1 token per atom for all atoms of the modified amino acid residue or nucleotide</li>
<li><strong>Glycan PTMs</strong>: 1 token per atom in the glycan (in addition to the 1 token for the residue the glycan is attached to)</li>
</ul>
<p>Input sequences are also limited to a minimum length of <strong>four</strong> characters.</p>
</section>
<section id="provide-input-sequences" class="level2">
<h2 class="anchored" data-anchor-id="provide-input-sequences">Provide Input Sequences</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Step 4
</div>
</div>
<div class="callout-body-container callout-body">
<p>Navigate to the input bar and click on <code>Add entity</code>. Make sure that both inputs have the <strong>protein</strong> molecule type. We will be inputing two amino acid sequences, one from rice and another one from its fungal pathogen <em>M. oryzae</em>.</p>
</div>
</div>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-5-contents" aria-controls="callout-5" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
View FASTA file
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-5" class="callout-5-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<div id="88e1d963" class="cell" data-execution_count="1">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="op">!</span> cat ..<span class="op">/</span>data<span class="op">/</span><span class="dv">8</span><span class="er">b2r</span>.fasta</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>&gt;8B2R_1|Chain A|Disease resistance protein RGA5|Oryza sativa (4530)
GPMRTKIVVKVHMPCGKSRAKAMALAASVNGVDSVEITGDDKDRLQVVGRGIDPVRLVALLREKCGLAELLQVEEVKE
&gt;8B2R_2|Chain B|AVR-Pik protein|Pyricularia oryzae (318829)
GPMRAIDLSRERDPNFFDNADIPVPECFWFMFKNNVRQDAGTCYSSWKMDKKVGPNWVHIKSDDNCNLSGDFPPGWIVLGKKRPGF</code></pre>
</div>
</div>
</div>
</div>
</div>
<p>In the <strong>first</strong> input, paste the following sequence:</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode sh code-with-copy"><code class="sourceCode bash"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="ex">GPMRTKIVVKVHMPCGKSRAKAMALAASVNGVDSVEITGDDKDRLQVVGRGIDPVRLVALLREKCGLAELLQVEEVKE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>In the <strong>second</strong> input, paste the following sequence:</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode sh code-with-copy"><code class="sourceCode bash"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="ex">GPMRTKIVVKVHMPCGKSRAKAMALAASVNGVDSVEITGDDKDRLQVVGRGIDPVRLVALLREKCGLAELLQVEEVKE</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Your input should look like the following:</p>
<p><img src="/assets/input.png" class="img-fluid"></p>
</section>
<section id="submit-job-and-view-results" class="level2 page-columns page-full">
<h2 class="anchored" data-anchor-id="submit-job-and-view-results">Submit Job and View Results</h2>
<div class="callout callout-style-default callout-tip callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Step 5
</div>
</div>
<div class="callout-body-container callout-body">
<p>Click <code>Continue and preview job</code> and wait for the job to finish. You will be redirected to the results page.</p>
</div>
</div>
<p>The results of your task will be stored as a hashed string on Google’s servers. Please take note of the identifier found in the URL to allow for easy sharing of predicted structures.</p>
<p><img src="/assets/id.png" class="img-fluid"></p>
<hr>
<p>The server will return the top five predicted structures, but will only display the highest scoring structure in the web interface. You can still access and view the other structures by clicking on the <strong>Download</strong> button.</p>
<p><img src="/assets/results.png" class="img-fluid"></p>
<p>To assess the quality of the predicted structure, AlphaFold Server also computes the following metrics:</p>
<table class="table-hover table">
<caption>Accuracy metrics to assess quality of predicted structures.</caption>
<colgroup>
<col style="width: 18%">
<col style="width: 18%">
<col style="width: 18%">
<col style="width: 43%">
</colgroup>
<thead>
<tr class="header">
<th style="text-align: center;">Metric</th>
<th style="text-align: center;">Range</th>
<th style="text-align: center;">Cutoff</th>
<th style="text-align: left;">Description</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center;">pLDDT</td>
<td style="text-align: center;">0-100</td>
<td style="text-align: center;">&gt; 70</td>
<td style="text-align: left;">per-atom confidence estimate</td>
</tr>
<tr class="even">
<td style="text-align: center;">PAE</td>
<td style="text-align: center;">[0, <span class="math inline">\(\infty\)</span>)</td>
<td style="text-align: center;">0-30</td>
<td style="text-align: left;">estimate of the error in the relative position and orientation between two atoms in the predicted structure</td>
</tr>
<tr class="odd">
<td style="text-align: center;">pTM / ipTM</td>
<td style="text-align: center;">0-1</td>
<td style="text-align: center;">&lt; 0.5</td>
<td style="text-align: left;">measures the accuracy of the entire structure based on nearest templates</td>
</tr>
</tbody>
</table>

<div class="no-row-height column-margin column-container"><div class="">
<p><strong>pLDDT</strong>: predicted local distance difference test</p>
<p><strong>PAE</strong>: predicted aligned error</p>
<p><strong>pTM</strong>: predicted template modeling score</p>
<p><strong>ipTM</strong>: interface predicted template modeling score</p>
</div></div></section>
</section>
<section id="json-format-for-running-jobs" class="level1">
<h1>JSON Format for Running Jobs</h1>
<p>Custom-made JSON files following a specific structure can be fed to the server for automation. We can see the JSON file used in our previous job by unzipping the output directory.</p>
<div id="4d8380cd" class="cell" data-execution_count="2">
<div class="sourceCode cell-code" id="cb5"><pre class="sourceCode python code-with-copy"><code class="sourceCode python"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a>json_path <span class="op">=</span> <span class="st">"../output/af3/fold_pik5_rga5_job_request.json"</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="op">!</span>cat {json_path} <span class="op">|</span> jq</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[
  {
    "name": "Pik5-RGA5",
    "modelSeeds": [
      "13"
    ],
    "sequences": [
      {
        "proteinChain": {
          "sequence": "GPMRTKIVVKVHMPCGKSRAKAMALAASVNGVDSVEITGDDKDRLQVVGRGIDPVRLVALLREKCGLAELLQVEEVKE",
          "count": 1
        }
      },
      {
        "proteinChain": {
          "sequence": "GPMRAIDLSRERDPNFFDNADIPVPECFWFMFKNNVRQDAGTCYSSWKMDKKVGPNWVHIKSDDNCNLSGDFPPGWIVLGKKRPGF",
          "count": 1
        }
      }
    ]
  }
]</code></pre>
</div>
</div>
<section id="job-name-seeds-and-sequences" class="level2">
<h2 class="anchored" data-anchor-id="job-name-seeds-and-sequences">Job name, seeds and sequences</h2>
<ul>
<li><code>name</code> is a string with the job name. This is how the job will appear as in the history table.</li>
<li><code>modelSeeds</code> is a list of strings of uint32 seed values (e.g.&nbsp;<code>["1827431", "122"]</code>). It is recommended to provide an empty list, which prompts the server to use a single random seed for the job.</li>
<li><code>sequences</code> is a list of dictionaries that carry descriptions of the molecules for modeling.</li>
</ul>
</section>
<section id="entity-types" class="level2">
<h2 class="anchored" data-anchor-id="entity-types">Entity types</h2>
<p>Valid entity types mirror those available in the AlphaFold Server web interface:</p>
<ul>
<li><code>proteinChain</code> - used for proteins</li>
<li><code>dnaSequence</code> - used for DNA (single strand)</li>
<li><code>rnaSequence</code> - used for RNA (single strand)</li>
<li><code>ligand</code> - used for allowed ligands</li>
<li><code>ion</code> - used for allowed ions</li>
</ul>
<p>Refer to the GitHub <a href="https://github.com/google-deepmind/alphafold/blob/main/server/README.md">documentation</a> for a complete list of fields for each entity type.</p>
</section>
<section id="json-structure" class="level2">
<h2 class="anchored" data-anchor-id="json-structure">JSON Structure</h2>
<center>
<div class="cell" data-layout-align="default">
<div class="cell-output-display">
<div>
<p></p><figure class="figure"><p></p>
<div>
<pre class="mermaid mermaid-js">flowchart TD
  root["JSON"]
  name["name"]
  seed["modelSeeds"]
  sequences["sequences"]

  root === name
  root === seed
  root ==== sequences

  sequences ----- protein["proteinChain"]
  sequences ----- dna["dnaSequence"]
  sequences --- rna["rnaSequence"]
  sequences --- lig["ligand"]
  sequences --- ion["ion"]

  protein --- seq["sequence"]
  protein --- count["count"]
  protein --- glycans["glycans"]
  protein --- mods["modifications"]

  glycans --- res["residues"]
  glycans --- pos["position"]

  mods --- type["ptmType"]
  mods --- pos2["position"]

  dna --- dnaseq["sequence"]
  dna --- dnacount["count"]
  dna --- dnamods["modifications"]

  dnamods --- dnamodtype["modificationType"]
  dnamods --- basepos["basePosition"]

  rna --- rnaseq["sequence"]
  rna --- rnacount["count"]
  rna --- rnamods["modifications"]

  lig --- lig2["ligand"]
  lig --- ligcount["count"]

  ion --- ion2["ion"]
  ion --- ioncount["count"]

</pre>
</div>
<p></p></figure><p></p>
</div>
</div>
</div>
</center>
</section>
</section>
<section id="frequently-asked-questions" class="level1">
<h1>Frequently Asked Questions</h1>
<div class="callout callout-style-default callout-caution no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-7-contents" aria-controls="callout-7" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
1. What biological molecule types can be modelled with AlphaFold Server?
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-7" class="callout-7-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The following biological molecules types are supported:</p>
<ul>
<li>Protein</li>
<li>DNA</li>
<li>RNA</li>
<li>Common ligands
<ul>
<li>ATP, ADP, AMP, GTP, GDP, FAD, NADP, NADPH, NDP</li>
<li>heme, heme C, myristic acid, oleic acid, palmitic acid, citric acid, chlorophylls A and B, bacteriophylls A and B</li>
</ul></li>
<li>Common ions
<ul>
<li>Ca<span class="math inline">\(^{2+}\)</span>, Co<span class="math inline">\(^{2+}\)</span>, Cu<span class="math inline">\(^{2+}\)</span>, Fe<span class="math inline">\(^{3+}\)</span>, K,<span class="math inline">\(^{+}\)</span> Mg<span class="math inline">\(^{2+}\)</span>, Mn<span class="math inline">\(^{2+}\)</span>, Na<span class="math inline">\(^{+}\)</span>, Zn<span class="math inline">\(^{2+}\)</span>, Cl<span class="math inline">\(^{-}\)</span></li>
</ul></li>
<li>Common post-translational modifications
<ul>
<li>Phosphorylation: serine, threonine, tyrosine, histidine</li>
<li>Acetylation: lysine</li>
<li>Methylation: lysine, arginine</li>
<li>Malonylation: cystein</li>
<li>Hydroxylation: proline, lysine, asparagine</li>
<li>Palmitoylation: cysteine</li>
<li>Succinylation: asparagine</li>
<li>S-nitrosylation: cysteine</li>
<li>Formylation: tryptophan</li>
<li>Crotonylation: lysine</li>
<li>Citrullination: lysine, arginine</li>
<li>Glycan chains composed of certain sugars:
<ul>
<li><span class="math inline">\(\alpha\)</span>-D-glucose, <span class="math inline">\(\beta\)</span>-D-glucose</li>
<li><span class="math inline">\(\alpha\)</span>-D-mannose, <span class="math inline">\(\beta\)</span>-D-mannose</li>
<li><span class="math inline">\(\alpha\)</span>-L-fucose</li>
<li><span class="math inline">\(\beta\)</span>-D-galactose</li>
<li>N-actyl-<span class="math inline">\(\beta\)</span>-D-glucosamine</li>
</ul></li>
</ul></li>
<li>Common chemical modifications of nucleic acids
<ul>
<li>DNA
<ul>
<li>Methylation: cytosine, guanine, adenine</li>
<li>Carboxylation: cytosine</li>
<li>Oxidation: guanine</li>
<li>Formylation: cytosine</li>
</ul></li>
<li>RNA
<ul>
<li>Methylation: cytosine, guanine, adenine, uracil</li>
<li>Isomerization: uridine <span class="math inline">\(\to\)</span> pseudouridine</li>
<li>Formylation: cytosine</li>
</ul></li>
</ul></li>
</ul>
</div>
</div>
</div>
<div class="callout callout-style-default callout-caution no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-8-contents" aria-controls="callout-8" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
2. What is the maximum job size allowed?
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-8" class="callout-8-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>5,000 tokens per job.</p>
</div>
</div>
</div>
<div class="callout callout-style-default callout-caution no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-9-contents" aria-controls="callout-9" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
3. How many jobs can I run on AlphaFold Server?
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-9" class="callout-9-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>20 jobs per day.</p>
</div>
</div>
</div>
<div class="callout callout-style-default callout-caution no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-10-contents" aria-controls="callout-10" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
4. How should I define inputs for AlphaFold Server?
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-10" class="callout-10-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p><strong>For Proteins</strong>: single-letter, standard amino-acid</p>
<p><strong>For DNA/RNA</strong>: single-letter, standard base</p>
<p><strong>For ligands, ions, and PTMs</strong>: three-letter CCD code as specified by <a href="https://www.ebi.ac.uk/pdbe-srv/pdbechem/">PDB</a></p>
</div>
</div>
</div>
<div class="callout callout-style-default callout-caution no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-11-contents" aria-controls="callout-11" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
5. What are seeds and how are they set?
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-11" class="callout-11-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>The model uses a ‘seed’ for internal random number generation. Normally this seed is sampled automatically, and will be resampled when cloning a job. Running multiple different seeds of the model and ranking over all the predictions can lead to improved accuracy. The seed used is saved into the output information per run.</p>
<p>To set a specific seed, turn off auto seed selection in the preview screen (after clicking the ‘Continue and Preview job’ button). The seed can be any integer between 0 and 4,294,967,295. When cloning a job where the seed was set, the seed will return to being automatically chosen by default.</p>
</div>
</div>
</div>
<div class="callout callout-style-default callout-caution no-icon callout-titled">
<div class="callout-header d-flex align-content-center" data-bs-toggle="collapse" data-bs-target=".callout-12-contents" aria-controls="callout-12" aria-expanded="false" aria-label="Toggle callout">
<div class="callout-icon-container">
<i class="callout-icon no-icon"></i>
</div>
<div class="callout-title-container flex-fill">
6. How can I increase the diversity of my predictions?
</div>
<div class="callout-btn-toggle d-inline-block border-0 py-1 ps-1 pe-0 float-end"><i class="callout-toggle"></i></div>
</div>
<div id="callout-12" class="callout-12-contents callout-collapse collapse">
<div class="callout-body-container callout-body">
<p>Run again with different seeds (it will be chosen automatically if not set). Users of AlphaFold2 have had success in generating diverse predictions by customizing MSA and/or template inputs to the model - this is not currently possible in the server but we hope to provide the ability to do similar customisations soon.</p>
</div>
</div>
</div>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>